### ISSUE TITLE
Container can't start after stop/restart.

### ISSUE THREAD

**Issue by yeta-io** [rancher/rancher#7193]
**Rancher Version:** 1.2.2

**Docker Version:** 1.12.3

**OS and where are the hosts located? (cloud, bare metal, etc):** Ubuntu Zesty

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

**Steps to Reproduce:**

Create stack, run container, then stop it and run again.

**Results:** 

Container immediately stopped after launch, the errors like below printed in console and displayed in UI

```
rancher_1  | 2016-12-23 20:29:05,857 ERROR [:] [] [] [] [cutorService-22] [.e.s.i.ProcessInstanceDispatcherImpl] Agent error for [storage.volume.remove.reply;agent=25]: 6 error(s) decoding:
rancher_1  |
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string'
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string'
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string'
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string'
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string'
rancher_1  | * 'Volume.Instance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string'
```

**Expected:**

Rainbow unicorns dancing when container launched successfully.

**Comment by tfiduccia** [comment]
@yeta-io I cannot reproduce this problem.  Can you please give more information.  What type of image are you using?  Can you give me the yaml files for that stack so I can try and reproduce it.

**Comment by yeta-io** [comment]
I'm using private docker images, so I can't share it with you.  Each image is based on ubuntu:devel and run java application (spring boot), the launch script (entrypoint) looks like `java -jar ....path to jar...`. Below the example version of docker file.

We are using HEALTHCHECK definition within the dockerfile itself, so maybe it the cause of problem.

```
FROM ubuntu:devel
MAINTAINER some maintainer <some@email.com>

RUN apt update \
&&  apt install -y curl \
&&  apt upgrade -y \
&&  apt clean;

HEALTHCHECK \
    --interval=1m \
    --timeout=3s \
    CMD \
    curl --fail http://127.0.0.1:80 || exit 1

EXPOSE 80
ENTRYPOINT java -jar /path/to/myapp.jar
```

**Comment by jpereiraio** [comment]
Same issue on my end (Upgrade from 1.1.2):

Rancher Version: 1.2.2
Docker Version: 1.12.2
OS and where are the hosts located? (cloud, bare metal, etc):  Ubuntu 14.04.4 LTS (3.13.0)
Setup Details: Rancher HA
Environment Type: Cattle
Steps to Reproduce:
Create stack, run container, then stop it and run again.

Results:
Container immediately stopped after launch, the errors like below printed in console and displayed in UI
```ERROR	service.activate.wait.exception	Waiting: container stopping [container:1i31611]: 10 error(s) decoding: 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[0].End' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[0].Start' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[1].End' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[1].Start' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[2].End' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[2].Start' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[3].End' expected a map, got 'string' 
* 'Instance.InstanceLinks[1].TargetInstance.Data.DockerInspect.State.Health.Log[3].Start' expected a map, got 'string' ```

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.3.0-rc6.
Created a service with image - sangeetha/testdhealth. 
It started successfully. Stop the container. container is able to start successfully without any issues.

Note - When testing with  rancher-server version - v1.3.0-rc3 using this same image - sangeetha/testdhealth , I was able to reproduce the same issue reported in the bug when stopping ( and starting) the container. It was not able to start successfully. 

Built image - sangeetha/testdhealth with the following dockerfile:

```
FROM ubuntu:devel
MAINTAINER some maintainer <some@email.com>

RUN apt update \
&&  apt install -y curl \
&&  apt upgrade -y \
&&  apt clean;

HEALTHCHECK \
    --interval=2s \
    --timeout=2s \
    --retries=1000 \
    CMD \
    curl --fail http://127.0.0.1:80 || exit 1

EXPOSE 80
```

@yeta-io , Thank you for sharing the dockerfile.

Fix for this issue will be available in rancher-server version - v1.3.0
